numpy/ma/mrecords.py

```diff
@@ -657,7 +657,7 @@ def openfile(fname):
     try:
         f = open(fname)
     except IOError:
-        raise IOError(f"No such file: '{fname}'")
+        raise IOError(f"No such file: '{fname}'") from None
```

Comment:
`from e` again; otherwise if the open fails for permission reasons, all we'll see is "no such file"